src/otg/l2g.py

```diff
@@ -145,7 +162,7 @@ def __post_init__(self: LocusToGeneStep) -> None:
                 study_locus=credible_set,
                 study_index=studies,
                 variant_gene=v2g,
-                # colocalisation=coloc,
+                colocalisation=coloc,
             )
 
             # Join and fill null values with 0
```

Comment:
Do we add 0 instead of Null in features?

Reply:
That's right. Most models can't handle missing values so you need to treat them somehow at the preprocessing step. This is what we used to do in production. However, XGBoost has a smart strategy to handle nulls and it's been in my "to try" list. XGBoost is like an ensemble of decision trees, and theoretically when there's a null in a node it evaluates the performance in each possible direction and it goes in the direction where the loss is smaller. 
![Screenshot_20240112-093558.png](https://github.com/opentargets/genetics_etl_python/assets/45119610/83ace41e-a7d4-47c1-bddc-dfaa037104b2)

I haven't experimented with this yet, so I can't estimate the impact. Implementation wise, it should be as easy as retraining without filling nulls. Should I give it a go?

A smarter way to go around nulls is to infer them. But I think we'd need different imputation strategies per feature.

Reply:
my previous experience with gradient boosting was as well to fill nulls with 0s. I see why the original L2G had to do the same and why this is our default strategy. Imputation is usually not recommended when the proportion of nulls is too big (as is our case for some features).

Exploring the impact of dealing differently with the nulls is potentially useful but goes beyond this PR. We can queue it with all the other L2G improvements that @addramir has in mind and revisit it after first release.